File: 1341-split-a-string-in-balanced-strings/split-a-string-in-balanced-strings.js
/**
 * @param {string} s
 * @return {number}
 */
var balancedStringSplit = function (s) {
    const n = s.length;
    let lCount = 0;
    let rCount = 0;

    let ans = 0;

    for (let i = 0; i < n; i++) {
        if (s[i] === 'L') {
            lCount++;
        } else {
            rCount++;
        }

        if (lCount === rCount) {
            ans++;
            lCount = 0;
            rCount = 0;
        }
    }

    return ans;
};

File: 1341-split-a-string-in-balanced-strings/split-a-string-in-balanced-strings.java
class Solution {
    public int balancedStringSplit(String s) {
        int lCount = 0;
        int rCount = 0;
        int ans = 0;

        for (int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);
            if (ch == 'L') {
                lCount++;
            } else {
                rCount++;
            }

            if (lCount == rCount) {
                ans++;
                lCount = 0;
                rCount = 0;
            }
        }

        return ans;
    }
}
